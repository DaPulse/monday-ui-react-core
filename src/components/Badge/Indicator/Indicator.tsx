import React from "react";
import cx from "classnames";
import styles from "./Indicator.module.scss";
import { IndicatorColor } from "./IndicatorConstants";
import { ComponentDefaultTestId } from "../../../tests/constants";

export interface IndicatorProps {
  color?: IndicatorColor;
}

const Indicator: React.FC<IndicatorProps> & { colors?: typeof IndicatorColor } = ({
  color = IndicatorColor.NOTIFICATION
}: IndicatorProps) => {
  return <div className={cx(styles.indicator, styles[color])} data-testid={ComponentDefaultTestId.INDICATOR} />;
};

Indicator.colors = IndicatorColor;

export default Indicator;
