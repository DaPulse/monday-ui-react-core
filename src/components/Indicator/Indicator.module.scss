@import "../../styles/themes.scss";

$indicator-size: 8px;

.indicator {
  width: $indicator-size;
  height: $indicator-size;
  border-radius: 50%;

  &.colorPrimary {
    background-color: var(--primary-color);
  }

  &.colorNotification {
    background-color: var(--negative-color);
  }
}
