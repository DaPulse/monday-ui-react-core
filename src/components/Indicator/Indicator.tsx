import React, { forwardRef, useRef } from "react";
import cx from "classnames";
import { VibeComponent, VibeComponentProps } from "../../types";
import styles from "./Indicator.module.scss";
import { getStyle } from "../../helpers/typesciptCssModulesHelper";
import { camelCase } from "lodash-es";
import { IndicatorColor } from "./IndicatorConstants";
import { getTestId } from "../../tests/test-ids-utils";
import { ComponentDefaultTestId } from "../../tests/constants";
import { useMergeRefs } from "../../hooks";

export interface IndicatorProps extends VibeComponentProps {
  indicatorColor?: IndicatorColor;
  ariaLabel?: string;
}

const Indicator: VibeComponent<IndicatorProps> & { colors?: typeof IndicatorColor } = forwardRef(
  (
    {
      indicatorColor = IndicatorColor.NOTIFICATION,
      ariaLabel,
      className,
      id,
      "data-testid": dataTestId
    }: IndicatorProps,
    ref
  ) => {
    const componentRef = useRef(null);
    const mergedRef = useMergeRefs({ refs: [ref, componentRef] });
    return (
      <div
        ref={mergedRef}
        className={cx(styles.indicator, getStyle(styles, camelCase("color-" + indicatorColor)), className)}
        aria-label={ariaLabel}
        id={id}
        data-testid={dataTestId || getTestId(ComponentDefaultTestId.INDICATOR, id)}
      />
    );
  }
);

Indicator.colors = IndicatorColor;

export default Indicator;
