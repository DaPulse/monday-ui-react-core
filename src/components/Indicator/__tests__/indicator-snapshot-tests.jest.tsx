import React from "react";
import renderer from "react-test-renderer";
import Indicator from "../Indicator";
import { IndicatorColor } from "../IndicatorConstants";

describe("Indicator", () => {
  it("renders correctly with empty props", () => {
    const tree = renderer.create(<Indicator />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with a custom indicatorColor", () => {
    const tree = renderer.create(<Indicator indicatorColor={IndicatorColor.PRIMARY} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("renders correctly with an ariaLabel", () => {
    const tree = renderer.create(<Indicator ariaLabel="test arialabel" />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
