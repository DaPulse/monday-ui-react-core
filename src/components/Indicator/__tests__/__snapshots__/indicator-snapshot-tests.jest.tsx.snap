// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Indicator renders correctly with a custom indicatorColor 1`] = `
<div
  className="indicator colorPrimary"
  data-testid="indicator"
/>
`;

exports[`Indicator renders correctly with an ariaLabel 1`] = `
<div
  aria-label="test arialabel"
  className="indicator colorNotification"
  data-testid="indicator"
/>
`;

exports[`Indicator renders correctly with empty props 1`] = `
<div
  className="indicator colorNotification"
  data-testid="indicator"
/>
`;
